Sources/PubNub/EventEngine/Subscribe/Effects/EmitMessagesEffect.swift

```diff
@@ -67,8 +67,8 @@ struct EmitMessagesEffect: EffectHandler {
       return false
     }
 
-    listeners.forEach {
-      $0.emit(batch: filteredMessages)
+    subscriptions.forEach {
```

Comment:
We could solve it by replacing the code you mentioned with the one below. What's your take on that?

```
 subscriptions.allObjects.forEach {
   $0.emit(batch: filteredMessages)
 }
```

Reply:
It can be, but memory and time will be wasted to creating a new list. Custom implementation should be pretty straightforward where we have the same method signature and use original `forEach` inside, but which call is wrapped into our `lockedRead` closure.

Reply:
I updated `WeakSet` so now it uses safe `forEach` method. The signature is the same and it still conforms to the `Sequence` protocol